api/v1beta1/types.go

```diff
@@ -125,11 +144,25 @@ type SubnetFilter struct {
 	CIDR            string `json:"cidr,omitempty"`
 	IPv6AddressMode string `json:"ipv6AddressMode,omitempty"`
 	IPv6RAMode      string `json:"ipv6RAMode,omitempty"`
-	ID              string `json:"id,omitempty"`
 
 	FilterByNeutronTags `json:",inline"`
 }
```

Comment:
I thought about that and I deliberately didn't do it because:
* It can easily be a follow-on
* The code-churn would be different to the code churn of this PR
* This is big enough already!